src/main/java/seedu/duke/data/Budget.java

```diff
@@ -53,6 +108,67 @@ public static String getBudgetLeft(long totalMonthlyExpense) {
         assert (Long.valueOf(MAX_AMOUNT_VALUE) * Long.valueOf(MAX_TRANSACTIONS_COUNT) > Long.valueOf(MAX_AMOUNT_VALUE));
         assert (MIN_BUDGET_VALUE > 0);
 
-        return Long.toString(budget - totalMonthlyExpense);
+        return budget - totalMonthlyExpense;
     }
+
+    /**
+     * Checks if the budget has been exceeded.
+     *
+     * <p>If the budget is negative, it means that the total monthly expense is higher than the budget left.
+     * Thus, the budget is exceeded and returns true.
+     *
+     * <p>If the budget is not positive, it means that the total monthly expense is lower than or equals to the
+     * budget left. Thus budget is not yet exceeded and returns false.
+     *
+     * @param budgetLeft A long value indicating the difference of total monthly expense and monthly budget.
+     * @return A boolean value indicating whether the budget has been exceeded for the month.
+     */
+    private static boolean hasExceededBudget(long budgetLeft) {
+        return budgetLeft < 0;
```

Comment:
Consider changing the `0` to a magic number, maybe like `PROHIBITED_BUDGET` or something for more meaning.